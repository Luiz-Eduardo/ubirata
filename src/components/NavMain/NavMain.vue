<script lang="ts" setup>
import NavItem from '@/components/NavItem/NavItem.vue';
import iconHome from '@/assets/icons/icon-home.png';
import iconLocation from '@/assets/icons/icon-location.svg';
</script>

<template>
  <nav class="bg-white-100 border-r-2 border-white-300 h-screen min-w-fit max-w-max">
    <ul>
      <NavItem
        to="/"
        :icon-src="iconHome"
        icon-alt-text="Home icon"
        title="Home"
      />
      <NavItem
        to="/map/neighborhoods"
        :icon-src="iconLocation"
        icon-alt-text="A location pin"
        title="Mapa"
      />
    </ul>
  </nav>
</template>
