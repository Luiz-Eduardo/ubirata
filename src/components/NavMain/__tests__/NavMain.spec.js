import '@testing-library/jest-dom';
import { render } from '@testing-library/vue';
import NavMain from '../NavMain.vue';

const renderComponent = (options) => render(NavMain, {...options });

const route = { fullPath: '/', path: '/' };

vi.mock('vue-router', () => ({
  useRoute: () => route,
  RouterLink: () => 'Home'
}));

describe('NavMain', () => {
  it('renders correctly', () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });
});
