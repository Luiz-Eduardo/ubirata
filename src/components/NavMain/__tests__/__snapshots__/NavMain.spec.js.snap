// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavMain > renders correctly 1`] = `
<div>
  <nav
    class="bg-white-100 border-r-2 border-white-300 h-screen min-w-fit max-w-max"
  >
    <ul>
      <li
        class="list-none"
      >
        Home
      </li>
      <li
        class="list-none"
      >
        Home
      </li>
    </ul>
  </nav>
</div>
`;
